File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/VelocityMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Velocity
import com.phamtunglam.health_connector_hc_android.pigeon.VelocityDto

/**
 * Converts a [VelocityDto] to a Health Connect [Velocity] object.
 */
internal fun VelocityDto.toHealthConnect(): Velocity = Velocity.metersPerSecond(metersPerSecond)

/**
 * Converts a Health Connect [Velocity] object to a [VelocityDto].
 *
 * Uses meters per second as the transfer unit for consistency.
 */
internal fun Velocity.toDto(): VelocityDto = VelocityDto(
    metersPerSecond = inMetersPerSecond,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/velocity_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Velocity;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show VelocityDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Velocity] to [VelocityDto].
@internal
extension VelocityToDto on Velocity {
  VelocityDto toDto() {
    // Uses meters per second as the transfer unit for consistency.
    return VelocityDto(
      metersPerSecond: inMetersPerSecond,
    );
  }
}

/// Converts [VelocityDto] to [Velocity].
@internal
extension VelocityDtoToDomain on VelocityDto {
  Velocity toDomain() {
    return Velocity.metersPerSecond(metersPerSecond);
  }
}
